src/extension/iSamples_resultList.js

```diff
@@ -34,7 +34,7 @@ class ResultList extends React.Component {
   // prevent unnecessary rerendering
   shouldComponentUpdate(nextProps) {
     this.switchView(store.getState()['query']['view']['facet']);
-    if (nextProps != this.props) {
+    if (nextProps !== this.props) {
```

Comment:
I am not confident if this is the right way to fix the circular dependency issue of `JSON.stringify`, probably will need to debug this issue and see if there is a better way than this.